src/ray/raylet/scheduling_policy.cc

```diff
@@ -1,32 +1,75 @@
 #include "scheduling_policy.h"
 
+#include "ray/util/logging.h"
+
 namespace ray {
 
+namespace raylet {
+
 SchedulingPolicy::SchedulingPolicy(const SchedulingQueue &scheduling_queue)
-    : scheduling_queue_(scheduling_queue) {}
+    : scheduling_queue_(scheduling_queue), gen_(rd_()) {}
 
 std::unordered_map<TaskID, ClientID, UniqueIDHasher> SchedulingPolicy::Schedule(
     const std::unordered_map<ClientID, SchedulingResources, UniqueIDHasher>
-        &cluster_resources) {
-  static ClientID local_node_id = ClientID::nil();
+        &cluster_resources,
+    const ClientID &local_client_id, const std::vector<ClientID> &others) {
+  // The policy decision to be returned.
   std::unordered_map<TaskID, ClientID, UniqueIDHasher> decision;
-  // TODO(atumanov): consider all cluster resources.
-  SchedulingResources resource_supply = cluster_resources.at(local_node_id);
-  const auto &resource_supply_set = resource_supply.GetAvailableResources();
+  // TODO(atumanov): protect DEBUG code blocks with ifdef DEBUG
+  RAY_LOG(DEBUG) << "[Schedule] cluster resource map: ";
+  for (const auto &client_resource_pair : cluster_resources) {
+    // pair = ClientID, SchedulingResources
+    const ClientID &client_id = client_resource_pair.first;
+    const SchedulingResources &resources = client_resource_pair.second;
+    RAY_LOG(DEBUG) << "client_id: " << client_id.hex() << " "
+                   << resources.GetAvailableResources().ToString();
+  }
 
   // Iterate over running tasks, get their resource demand and try to schedule.
   for (const auto &t : scheduling_queue_.GetReadyTasks()) {
     // Get task's resource demand
     const auto &resource_demand = t.GetTaskSpecification().GetRequiredResources();
-    bool task_feasible = resource_demand.IsSubset(resource_supply_set);
-    if (task_feasible) {
-      const TaskID &task_id = t.GetTaskSpecification().TaskId();
-      decision[task_id] = local_node_id;
+    const TaskID &task_id = t.GetTaskSpecification().TaskId();
+    RAY_LOG(DEBUG) << "[SchedulingPolicy]: task=" << task_id.hex()
```

Comment:
don't do `<< task_id.hex()`, just `task_id` this applies in many places (to other kinds of IDs as well)